src/Gui/StyleParameters.cpp

```diff
@@ -0,0 +1,801 @@
+            pos += 2;
+            int g = std::stoi(input.substr(pos, 2), nullptr, hexadecimalBase);
+            pos += 2;
+            int b = std::stoi(input.substr(pos, 2), nullptr, hexadecimalBase);
+            pos += 2;
+
+            return std::make_unique<Color>(QColor(r, g, b));
+        };
+
+        const auto parseFunctionStyleColor = [&]() {
+            bool hasAlpha = peekString(rgbaFunction);
+
+            pos += hasAlpha ? strlen(rgbaFunction) : strlen(rgbFunction);
+
+            int r = parseInt();
+            if (!match(',')) {
+                throw std::runtime_error("Expected ',' after red");
+            }
+            int g = parseInt();
+            if (!match(',')) {
+                throw std::runtime_error("Expected ',' after green");
+            }
+            int b = parseInt();
+            int a = 255; // NOLINT(*-magic-numbers)
+            if (hasAlpha) {
+                if (!match(',')) {
+                    throw std::runtime_error("Expected ',' after blue");
+                }
+                a = parseInt();
+            }
+            if (!match(')')) {
+                throw std::runtime_error("Expected ')' after color arguments");
+            }
+            return std::make_unique<Color>(QColor(r, g, b, a));
+        };
+
+        skipWhitespace();
+
+        if (input[pos] == '#') {
+            return parseHexadecimalColor();
+        }
+
+        if (peekString(rgbFunction) || peekString(rgbaFunction)) {
+            return parseFunctionStyleColor();
+        }
+
+        throw std::runtime_error("Unknown color format");
+    }
+
+
+    bool peekParameter()
+    {
+        skipWhitespace();
+        return pos < input.size() && input[pos] == '@';
+    }
+
+    std::unique_ptr<Expr> parseParameter()
+    {
+        skipWhitespace();
+        if (!match('@')) {
+            throw std::runtime_error("Expected '@' for parameter");
+        }
+        size_t start = pos;
+        while (pos < input.size() && (isalnum(input[pos]) || input[pos] == '_')) {
+            ++pos;
+        }
+        if (start == pos) {
+            throw std::runtime_error("Expected parameter name after '@'");
+        }
+        return std::make_unique<ParameterReference>(input.substr(start, pos - start));
+    }
+
+    bool peekFunction()
+    {
+        skipWhitespace();
+        return pos < input.size() && isalpha(input[pos]);
+    }
+
+    std::unique_ptr<Expr> parseFunctionCall()
+    {
+        skipWhitespace();
+        size_t start = pos;
+        while (pos < input.size() && isalnum(input[pos])) {
+            ++pos;
+        }
+        std::string functionName = input.substr(start, pos - start);
+
+        if (!match('(')) {
+            throw std::runtime_error("Expected '(' after function name");
+        }
+
+        std::vector<std::unique_ptr<Expr>> arguments;
+        if (!match(')')) {
+            do {
+                arguments.push_back(parseExpression());
+            } while (match(','));
```

Comment:
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es75-avoid-do-statements
```suggestion
            arguments.push_back(parseExpression());
            while (match(',')) {
                arguments.push_back(parseExpression());
            }
```

or just suppress it here, I think this is one of the few cases where do-while is clear enough to avoid issues.

Reply:
I'll supress it here - changing it into while requires duplicating logic which is not fine I think